netaddr.go

```diff
@@ -720,7 +732,38 @@ func (ip IP) string6() string {
 			ret = append(ret, ':')
 		}
 
-		ret = appendHex(ret, ip.v6u16(i))
+		ret = appendHex(ret, ip.v6u16(i), noPadHex)
+	}
+
+	if ip.z != z6noz {
+		ret = append(ret, '%')
+		ret = append(ret, ip.Zone()...)
+	}
+	return string(ret)
+}
+
+// StringExpanded returns the string form of the IP address ip, but IPv6
+// addresses are expanded with leading zeroes and no "::" compression. For
+// example, "2001:db8::1" becomes "2001:0db8:0000:0000:0000:0000:0000:0001".
+//
+// IPv4 and invalid IP addresses are formatted the same as when using IP.String.
+// See the documentation of IP.String for more details.
+func (ip IP) StringExpanded() string {
+	// Invalid and IPv4 are handled the same as in String.
+	switch ip.z {
+	case z0, z4:
+		return ip.String()
+	}
+
+	// See the comment in IP.string6.
+	const max = len("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff%enp5s0")
```

Comment:
why guess on this length? we know the exact length this will be, as we know the Zone (pull it up from the bottom)